src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/EverythingOperationContinuationToken.cs

```diff
@@ -3,42 +3,88 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
-using System.Text.Json;
+using System.Collections.Generic;
+using Newtonsoft.Json;
 
 namespace Microsoft.Health.Fhir.Core.Features.Operations.Everything
 {
-    public class EverythingOperationContinuationToken
+    internal class EverythingOperationContinuationToken
     {
-        public EverythingOperationContinuationToken(int phase, string internalContinuationToken)
+        [JsonConstructor]
+        internal EverythingOperationContinuationToken(int phase, string internalContinuationToken)
         {
             Phase = phase;
             InternalContinuationToken = internalContinuationToken;
+
+            SeeAlsoLinks = new List<string>();
+            CurrentSeeAlsoLinkIndex = -1;
         }
 
-        public int Phase { get; private set; }
+        [JsonProperty]
+        internal List<string> SeeAlsoLinks { get; }
+
+        [JsonProperty]
+        internal int Phase { get; set; }
+
+        [JsonProperty]
+        internal string InternalContinuationToken { get; set; }
 
-        public string InternalContinuationToken { get; private set; }
+        internal bool MoreSeeAlsoLinksToProcess
+        {
+            get
+            {
+                return CurrentSeeAlsoLinkIndex < SeeAlsoLinks.Count - 1;
+            }
+        }
 
-        public static string ToString(int phase, string internalContinuationToken)
+        internal string CurrentSeeAlsoLinkId
         {
-            return JsonSerializer.Serialize(new EverythingOperationContinuationToken(phase, internalContinuationToken));
+            get
+            {
+                return SeeAlsoLinks[CurrentSeeAlsoLinkIndex];
```

Comment:
We should check the index before accessing. Otherwise exceptions will be thrown when `CurrentSeeAlsoLinkIndex` is the default value -1.

Reply:
Added some safety checks in a9f916c.